projects/igniteui-angular/src/lib/query-builder/query-builder.component.ts

```diff
@@ -53,10 +54,26 @@ export class IgxQueryBuilderComponent implements OnDestroy {
     public showEntityChangeDialog = true;
 
     /**
-     * Gets/sets the entities.
+     * Returns the entities.
+     * @hidden
+     */
+    public get entities(): EntityType[] {
+        return this._entities;
+    }
+
+    /**
+     * Sets the entities.
+     * @hidden
      */
     @Input()
-    public entities: EntityType[];
+    public set entities(entities: EntityType[]) {
+        this._entities = entities;
+        if (JSON.stringify(entities) !== JSON.stringify(this._entities)) {
```

Comment:
`JSON.stringify(entities)` can fail with circular ref error if `fields` are actual column instances provided from the grid (advanced filter)

Reply:
PS: Also above `this._entities = entities;` 👀